### PR TITLE
add SignedContingentInput proto conversion

### PR DESCRIPTION
**Pull request by eranrund**
### Motivation

We need a way to go between the Rust/prost struct and the protobuf version. This is similar to a whole bunch of other conversion code we already have.

### CODE REVIEW

File: api/src/convert/signed_contingent_input.rs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    fn try_from(src: &external::SignedContingentInput) -> Result<Self, Self::Error> {
+        let block_version = src.get_block_version();
+        let tx_in = TxIn::try_from(src.get_tx_in())?;
+        let mlsag = RingMLSAG::try_from(src.get_mlsag())?;
+        let pseudo_output_amount = UnmaskedAmount::try_from(src.get_pseudo_output_amount())?;
+        let required_output_amounts = src
+            .get_required_output_amounts()
+            .iter()
+            .map(UnmaskedAmount::try_from)
+            .collect::<Result<Vec<UnmaskedAmount>, ConversionError>>()?;
```
Comment (nick-mobilecoin) [inline api/src/convert/signed_contingent_input.rs:44]: :upside_down_face: : I have no objection to leaving this explicit.  I generally omit (probably  me being lazy)
```suggestion
            .collect::<Result<Vec<_>, _>>()?;
```

File: api/src/convert/signed_contingent_input.rs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+        // Converting mc_transaction_extra::SignedContingentInput ->
+        // external::SignedContingentInput ->
+        // mc_transaction_extra::SignedContingentInput should be the identity
+        // function.
+        {
+            let external_sci: external::SignedContingentInput =
+                external::SignedContingentInput::from(&sci);
+            let recovered_sci: SignedContingentInput =
+                SignedContingentInput::try_from(&external_sci).unwrap();
```
Comment (nick-mobilecoin) [inline api/src/convert/signed_contingent_input.rs:139]: :upside_down_face:  take it or leave it
```suggestion
            let external_sci = external::SignedContingentInput::from(&sci);
            let recovered_sci = SignedContingentInput::try_from(&external_sci).unwrap();
```

File: api/src/convert/signed_contingent_input.rs [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        {
+            let bytes = mc_util_serial::encode(&sci);
+            let recovered_sci = external::SignedContingentInput::parse_from_bytes(&bytes).unwrap();
+            assert_eq!(recovered_sci, external::SignedContingentInput::from(&sci));
+        }
+
+        // Encoding with protobuf, decoding with prost should be the identity function.
+        {
+            let external_sci: external::SignedContingentInput =
+                external::SignedContingentInput::from(&sci);
```
Comment (nick-mobilecoin) [inline api/src/convert/signed_contingent_input.rs:153]: :upside_down_face: 
```suggestion
            let external_sci = external::SignedContingentInput::from(&sci);
```

File: api/src/convert/signed_contingent_input.rs [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+            let bytes = external_sci.write_to_bytes().unwrap();
+            let recovered_sci: SignedContingentInput = mc_util_serial::decode(&bytes).unwrap();
```
Comment (nick-mobilecoin) [inline api/src/convert/signed_contingent_input.rs:155]: :upside_down_face: I'm kind of surprised this one can be omitted, must be eliding type by the `assert_eq` below
```suggestion
            let recovered_sci = mc_util_serial::decode(&bytes).unwrap();
```

**Review by nick-mobilecoin — APPROVED** [review]
LGTM some minor nits/suggested tweaks

**Review by jcape — APPROVED** [review]

**Pull request merged**